pay-api/src/pay_api/resources/invoice.py

```diff
@@ -11,38 +11,92 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Resource for Invoice endpoints."""
+"""Resource for Payment endpoints."""
 from http import HTTPStatus
 
-from flask import jsonify
+from flask import current_app, jsonify, request
 from flask_restplus import Namespace, Resource, cors
 
-from pay_api.exceptions import BusinessException
-from pay_api.services import InvoiceService
+from pay_api.exceptions import error_to_response, BusinessException, ServiceUnavailableException
+from pay_api.schemas import utils as schema_utils
+from pay_api.services import PaymentService, InvoiceService
+from pay_api.services.auth import check_auth
 from pay_api.utils.auth import jwt as _jwt
+from pay_api.utils.constants import EDIT_ROLE
 from pay_api.utils.trace import tracing as _tracing
-from pay_api.utils.util import cors_preflight
+from pay_api.utils.errors import Error
+from pay_api.utils.util import cors_preflight, get_str_by_path
 
-API = Namespace('invoices', description='Payment System - Payment Requests')
 
+API = Namespace('invoice', description='Payment System - Invoices')
 
-@cors_preflight(['GET'])
-@API.route('', methods=['GET', 'OPTIONS'], doc={'deprecated': True})
-class PaymentRequestInvoice(Resource):
-    """Temporary endpoint to unblock teams who are using this endpoint."""
+
```

Comment:
This file has been renamed from payment to invoice.py.. diff doesn't reflect this.